repo/repo.go

```diff
@@ -69,6 +70,9 @@ type Repo interface {
 	// Repos have a logbook for recording & storing operation logs
 	Logbook() *logbook.Book
 
+	// Dscache is a cache of datasets that have been built according to logbook
+	Dscache() *dscache.Dscache
```

Comment:
Can we move this closer to refstore, and add a note to refstore mentioning that we're transitioning toward using dscache?